plaso/parsers/gdrive_synclog.py

```diff
@@ -0,0 +1,197 @@
+  _TWO_DIGITS = text_parser.PyparsingConstants.TWO_DIGITS
+
+  # Ignore commas and spaces.
+  IGNORE_FIELD = pyparsing.CharsNotIn(',').suppress()
+
+  _GDS_DATE_TIME = (
+      _FOUR_DIGITS.setResultsName('year') + _HYPHEN +
+      _TWO_DIGITS.setResultsName('month') + _HYPHEN +
+      _TWO_DIGITS.setResultsName('day') +
+      text_parser.PyparsingConstants.TIME_MSEC_ELEMENTS +
+      pyparsing.Word(pyparsing.printables).setResultsName('time_zone_offset')
+  ).setResultsName('date_time')
+
+  # Multiline entry end marker, matched from right to left.
+  _GDS_ENTRY_END = pyparsing.StringEnd() | _GDS_DATE_TIME
+
+  _GDS_LINE = (
+      _GDS_DATE_TIME +
+      pyparsing.Word(pyparsing.alphas).setResultsName('log_level') +
+      # TODO: strip pid= out, cast to integers?
+      pyparsing.Word(pyparsing.printables).setResultsName('pid') +
+      # TODO: consider stripping thread identifier/cleaning up thread name?
+      pyparsing.Word(pyparsing.printables).setResultsName('thread') +
+      pyparsing.Word(pyparsing.printables).setResultsName('source_code') +
+      pyparsing.SkipTo(_GDS_ENTRY_END).setResultsName('message') +
+      pyparsing.ZeroOrMore(pyparsing.lineEnd()))
+
+  LINE_STRUCTURES = [
+      ('logline', _GDS_LINE),
+  ]
+
+  def _ParseLine(self, parser_mediator, structure):
+    """Parses a logline and store appropriate attributes.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          a line of a text file.
+    """
+    # Some hacks follow: our log source uses a different datetime format than
+    # dfdatetime does, and there appear to be some pending changes surrounding
+    # is_local_time and converting to UTC (see
+    # https://github.com/log2timeline/dfdatetime/issues/47, possibly others).
+    # As such, I'm hacking around things with some string reformats to get it
+    # working for a first-pass, and will take direction from Joachim when I have
+    # more context on the ongoing work/best practices.
+    year, month, day_of_month, hours, minutes, seconds, milliseconds, tz = (
+        structure.date_time)
+
+    # TimeElementsInMilliseconds appears not to support local timezones, yet,
+    # and time_elements.CopyFromDateTimeString requires a different format.
+    # Initialize it without the TZ, then convert to a TZ-aware value with
+    # a formatting hack and CopyFromDateTimeString.
+    dfdatetime_obj = dfdatetime_time_elements.TimeElementsInMilliseconds(
+        (year, month, day_of_month, hours, minutes, seconds, milliseconds))
+
+    dt_hack = ('{0:d}-{1:02d}-{2:02d} {3:02d}:'
+               '{4:02d}:{5:02d}.{6:03d}{7!s}:{8!s}').format(
+                   year, month, day_of_month, hours, minutes, seconds,
+                   milliseconds, tz[0:3], tz[3:5])
+    try:
+      dfdatetime_obj.CopyFromDateTimeString(dt_hack)
```

Comment:
I opt to use CopyFromStringISO8601 instead

Reply:
Done, and revised/cleaned up some of the variable names/comments when I reshuffled this section into another method, PTAL.